tests/c/interp_inside_call2.c

```diff
@@ -12,6 +12,7 @@
 //     jit-state: enter-stopgap
 //     ...
 //     jit-state: exit-stopgap
+//     jit-state: exit-jit-code
```

Comment:
Should the `jit-state: enter-stopgap` and `jit-state: exit-stopgap` states be replaced by a single `jit-state: deoptimise`?

Reply:
Not sure what advantage this would have. We know we are deoptimising when we see `enter-stopgap/exit-stopgap`.

Reply:
But we don't stopgap, do we?

Reply:
I misunderstood you. You want to rename it. I guess we can. Is what we are doing now not stopgapping anymore? Just looking at the definition of stopgap I would still say it fits.

Reply:
Regardless of what you call it, whereas before stopgapping was an interval of time (in which we happened to be running an interpreter), now it's an instantaneous event. We flick from JITted code back to the interpreter without any intermediate "user program execution". So to start/stop stopgapping makes little sense in my head.

And I do still think it should be called "deoptimise" and not "stopgap". The name "stopgap" implies a temporary compromise, which no longer conveys what we are really doing any more.

Reply:
Edd's summed up my position well.

Reply:
I'm not attached to stopgapping and get your point. So I'm happy to rename it.

Reply:
Correct: "guard failure" / "deoptimisation" (they're the same thing for us) is effectively a discreet event.

Reply:
Well this wasn't fun. [303a233](https://github.com/ykjit/yk/pull/578/commits/303a23327920839ec2c4f03f89b9c7c455029e3d)

Reply:
But now it's done :)